### PR TITLE
Performance fix for enrollment roster api.

### PR DESCRIPTION
**Pull request by dcadams**
This reduces the number of selects from an unscalable n+1 where n is a function of the number of students enrolled to 2.

Although there is no coded test for this, at this time, it has been fully tested on my system and comes with a money back guarantee.

@stvstnfrd @jbassen

### CODE REVIEW

File: common/djangoapps/enrollment/data.py [hunk 1]
```diff
@@ -298,27 +295,51 @@ def get_course_enrollment_info(course_id, include_expired=False):
...
+        'user__username',
+        'user__email',
+        'mode',
+        'user__profile__name',
+    ).order_by(
+        'user__username'
+    )
+
+    # Find all user_ids with instructor or staff roles in course.
+    non_students = CourseAccessRole.objects.filter(
```
Comment (stvstnfrd) [inline common/djangoapps/enrollment/data.py]: Why are we querying this independently?
Could we include it in the `JOIN` above?

Ostensibly, we could then drop the inner `for non_student in non_students` loop below.

Comment (dcadams) [inline common/djangoapps/enrollment/data.py]: I decided to do it this way as I've never seen a course with more than 4 or 5 staff so the separate sql will return only that number of rows. In this case I think including another join for this in the first sql would be inefficient as the vast majority wouldn't match.

Comment (stvstnfrd) [inline common/djangoapps/enrollment/data.py]: Makes sense.

File: common/djangoapps/enrollment/data.py [hunk 2]
```diff
@@ -298,27 +295,51 @@ def get_course_enrollment_info(course_id, include_expired=False):
...
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user_id').distinct()
 
+    roster = []
+    for enrollment in enrollments:
+        is_staff = 0
+        for non_student in non_students:
```
Comment (stvstnfrd) [inline common/djangoapps/enrollment/data.py]: @dcadams This is fine as-is, but just as "trivia", here's a more Pythonic approach:

```python
is_staff = any(True for x in non_students if x.values()[0] == enrollment['user_id'])
```

**Pull request merged**